src/viking.js

```diff
@@ -1,14 +1,92 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength){
+    this.health = health;
+    this.strength = strength;
+  }
+  attack(){
+    return this.strength;
+  }
+  receiveDamage(damage){
+    this.strength -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength){
+    super(health, strength);
+    this.name = name;
+  }
 
+  receiveDamage(damage){
+    this.health = this.health - damage;
+    if (this.health > 0){
+      return `${this.name} has received ${damage} points of damage`;
+    } else {
+      return `${this.name} has died in act of combat`;
+    }
+  }
+  battleCry(){
+    return "Odin Owns You All!";
+  }
+}
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+
+receiveDamage(damage){
+  this.health -= damage;
+  if (this.health <= 0){
+    return "A Saxon has died in combat";
+  }
+  return `A Saxon has received ${damage} points of damage`;
+}
+}
 
 // War
-class War {}
+class War {
+  constructor(){
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+
+  }
+  addViking(vikingObject){
+    this.vikingArmy.push(vikingObject);
+  }
+  addSaxon(saxonObject){
+    this.saxonArmy.push(saxonObject);
+  }
+  vikingAttack(){
+
+    if (this.vikingArmy.length === 0 || this.saxonArmy.length === 0){
+      return "Both armies needs soldiers to batle agains each other";
+    }
+
+    const vikingAndSaxonMin = 0;
+    const vikingMax = this.vikingArmy.length;
+    const saxonMax = this.saxonArmy.length;
+
+    const vikingIndex = Math.floor(Math.random() * (vikingMax - vikingAndSaxonMin)) + vikingAndSaxonMin;
+    const saxonIndex = Math.floor(Math.random() * (saxonMax - vikingAndSaxonMin)) + vikingAndSaxonMin;
+
```

Comment:
aqui vc poderia ter feito direto Math.floor(Math.random() * this.vikingArmy.length) 👍